File: in_Java/0258 Add Digits.java
class Solution {
    public int addDigits(int num) {
        while(String.valueOf(num).length() != 1){
            int cur = 0;
            while(num != 0){
                cur += num % 10;
                num /= 10;
            }
            num = cur;
        }
        return num;
    }
}

File: in_Python/0258 Add Digits.py
class Solution(object):
    def dsum (self, n):
        res = 0
        while n != 0:
            res += n % 10
            n /= 10
        return res
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        while num >= 10:
            num = self.dsum(num)
        return num
        
        